### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by BudnikOleksii**
- [DEMO LINK](https://BudnikOleksii.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by mgrinko — CHANGES_REQUESTED** [review]
I tried to toggle `JS` option but for some reason the loader appeared at the bottom

Also the input became focused which is wrong I think

https://prnt.sc/Hxs7-FcfMtFX

**Comment by BudnikOleksii** [comment]
> I tried to toggle `JS` option but for some reason the loader appeared at the bottom
> 
> Also the input became focused which is wrong I think
> 
> https://prnt.sc/Hxs7-FcfMtFX

When input changes status active/completed I change it on the server and update it accordingly

**Comment by BudnikOleksii** [comment]
> I tried to toggle `JS` option but for some reason the loader appeared at the bottom
> 
> https://prnt.sc/Hxs7-FcfMtFX
> 
> * Also the input became focused which is wrong I think
> * Filters should not jump when Clear Completed button disappears (see the demo)

Thank you, fixed

**Comment by mgrinko** [comment]
- Loaders should be disabled in case of error
- In case of update error title should not be changed

**Comment by BudnikOleksii** [comment]
> * Loaders should be disabled in case of error
> * In case of update error title should not be changed

My bad, thank you. Also, fixed a bug when the title is empty and a delete error occurred

File: src/api/todos.ts [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
 export const getTodos = (userId: number) => {
   return client.get<Todo[]>(`/todos?userId=${userId}`);
 };
 
-// Add more methods here
+export const addTodo = (todo: NewTodo): Promise<Todo> => {
+  return client.post<Todo>('/todos', todo);
+};
+
+export const removeTodoByTodoId = (todoId: number) => {
```
Comment (VETALEK) [inline src/api/todos.ts:12]:
```suggestion
export const removeTodoById = (todoId: number) => {
```

File: src/api/todos.ts [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+  return client.delete(`/todos/${todoId}`);
+};
+
+export const updateTodoByTodoId = (
```
Comment (VETALEK) [inline src/api/todos.ts:16]:
```suggestion
export const updateTodoById = (
```

File: src/components/TodoFooter/TodoFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    todos.filter(todo => todo.completed)
+  ), [todos]);
+
+  const handleRemoveCompletedTodos = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        handleRemoveTodo(todo.id);
+      }
+    });
+  };
```
Comment (VETALEK) [inline src/components/TodoFooter/TodoFooter.tsx:31]: let's wrap it with `useCallback`

Comment (VETALEK) [reply src/components/TodoFooter/TodoFooter.tsx:31]: let's do it with `Promise.all` (i believe removing todo is async function)

image

**Review by VETALEK — CHANGES_REQUESTED** [review]
GJ 🔥 
commented

File: src/components/TodoFooter/TodoFooter.tsx (hunk 3 above)
Comment (BudnikOleksii) [reply src/components/TodoFooter/TodoFooter.tsx]: > let's do it with `Promise.all` (i believe removing todo is async function)
> 
> image

I had done with Promise.all, but then found that it is possible to handle with other methods😁 I Refactored one more time.
Also, I think it is unnecessary to wrap with useCallback if the function I use only in the same component where it is declared, but ok.

**Review by VETALEK — APPROVED** [review]
🚀🚀🚀

**Pull request closed without merging**